test/plugin/test_out_elasticsearch.rb

```diff
@@ -974,14 +1021,17 @@ def test_reconnect_on_error_enabled
     end
```

Comment:
test_reconnection_on_error_enabled test case is affected to test driver behavior change.

I tried to adopt this test case for v0.14 test driver as follows:

```ruby
  def test_reconnect_on_error_enabled
    connection_resets = 0

    stub_elastic_ping(url="http://localhost:9200").with do |req|
      connection_resets += 1
    end

    stub_request(:post, "http://localhost:9200/_bulk").with do |req|
      raise ZeroDivisionError, "any not host_unreachable_exceptions exception"
    end

    driver.configure("reconnect_on_error true\n")

    assert_raise(ZeroDivisionError) {
      driver.run(default_tag: 'test', flush: true, wait_flush_completion: false, force_flush_retry: false, shutdown: false) do
        driver.feed(sample_record)
      end
    }

    driver.run(default_tag: 'test', flush: true, wait_flush_completion: false, force_flush_retry: true, shutdown: false) do
      driver.feed(sample_record)
    end

    # XXX: connection_resets should be 2. How to fix it?
    assert_equal(connection_resets, 1)
  end
```

But I could not obtain same value of `connection_resets` in v0.12.
How do we treat this test case?